### ISSUE TITLE
🐛 Type detected as an unused variable.

### ISSUE THREAD

**Issue by NikolaRHristov** [biomejs/biome#523]
### Environment information

```block
CLI:
  Version:                      1.2.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.8.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0

Discovering running Biome servers...

Incompatible Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

i Rage discovered this running server using an incompatible version of Biome.

Server:
  Version:                      1.2.2-nightly.9264c09

Running Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

i The client isn't connected to any server but rage discovered this running Biome server.

Server:
  Version:                      1.2.2
  Name:                         biome_lsp
  CPU Architecture:             x86_64
  OS:                           windows

Workspace:
  Open Documents:               0

Other Active Server Workspaces:

Workspace:
  Open Documents:               2
  Client Name:                  Visual Studio Code - Insiders
  Client Version:               1.84.0-insider
```

### What happened?

The type is being used as `satisfies X`

<img width="622" alt="image" src="https://github.com/biomejs/biome/assets/90222871/45043a2b-7e5e-494e-8f51-ea922e8b49c4">

but it's coming up as an unused variable

<img width="660" alt="image" src="https://github.com/biomejs/biome/assets/90222871/c75d1f57-89f7-459f-a307-781341822570">

### Expected result

To properly detect `satisfies *Type*`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by NikolaRHristov** [comment]
It works for normal Type association 

<img width="554" alt="image" src="https://github.com/biomejs/biome/assets/90222871/46472cb0-dd98-4b1b-b207-c8b9ed5bc23a">

<img width="602" alt="image" src="https://github.com/biomejs/biome/assets/90222871/e7facbe9-15cd-45f6-91c7-e6c6ce76fc14">

**Comment by ematipico** [comment]
Can you check if you have the same issue in the playground?

**Comment by NikolaRHristov** [comment]
No, it works properly or at least doesn't show any errors https://biomejs.dev/playground/?indentWidth=4&files.type.ts=ZQB4AHAAbwByAHQAIABkAGUAZgBhAHUAbAB0ACAAaQBuAHQAZQByAGYAYQBjAGUAIABUAHkAcABlACAAewAKACAAIAB0AGUAcwB0ADoAIABuAHUAbQBiAGUAcgA7AAoAIAAgAHQAZQBzAHQAMgA6ACAAcwB0AHIAaQBuAGcAOwAKAH0AOwA%3D&files.main.tsx=ZQB4AHAAbwByAHQAIABkAGUAZgBhAHUAbAB0ACAAKABUAGUAcwB0ADoAIABUAHkAcABlADIAKQAgAD0APgAgAHsACgAgACAAYwBvAG4AcwB0ACAAVABlAHMAdAAyACAAPQAgAHsACgAgACAAIAAgAHQAZQBzAHQAOgAgADIALAAKACAAIAAgACAAdABlAHMAdAAyADoAIAAnAHQAZQBzAHQAJwAKACAAIAB9ACAAcwBhAHQAaQBzAGYAaQBlAHMAIABUAHkAcABlADsACgAKACAAIAByAGUAdAB1AHIAbgAgAFQAZQBzAHQAMgA7AAoAfQAKAAoAaQBtAHAAbwByAHQAIAB0AHkAcABlACAAVAB5AHAAZQAyACAAZgByAG8AbQAgACIAdAB5AHAAZQAuAHQAcwAiADsACgAKAGkAbQBwAG8AcgB0ACAAdAB5AHAAZQAgAFQAeQBwAGUAIABmAHIAbwBtACAAIgB0AHkAcABlAC4AdABzACIAOwA%3D

**Comment by ematipico** [comment]
Great, you'll get the fix in the next release

**Comment by NikolaRHristov** [comment]
Can confirm this was fixed with `@biomejs/biome` `v1.3.0` https://github.com/biomejs/biome/pull/544
